bson.cpp

```diff
@@ -1064,6 +1145,49 @@ void parseTypeMap(hippo_bson_conversion_options_t *options, const Array &typemap
 			options->array_class_name = array_type;
 		}
 	}
+
+	if (typemap.exists(s_types)) {
+		if (!typemap[s_types].isArray()) {
+			throw MongoDriver::Utils::throwInvalidArgumentException("The 'types' key in the type map should be an array");
+		}
+
+		for (ArrayIter iter(typemap[s_types]); iter; ++iter) {
+			Variant key(iter.first());
+			String s_key = key.toString();
+
+			const Variant& data(iter.secondRef());
+			if (!data.isString()) {
+				throw MongoDriver::Utils::throwInvalidArgumentException("The typemap for type '" + s_key + "' should be a string");
+			}
+			String s_type_class = data.toString();
+
+			/* Depending on the key, set the right option (or throw an exception) */
+			if (CASECMP(s_key, s_MongoBsonBinary_shortName)) {
```

Comment:
Are you still considering TypeWrappers for the basic types? If so, how would you handle those?

I might consider replacing the short names with similarly named constants whose values are the actual BSON type integers, but that might be problematic for the integer and Javascript types, which each have two variations (32/64-bit and optional scopes) that we'd like to handle the same.